### PR TITLE
Implement Regex based title cleaning

### LINKED ISSUE #412: Extras - show titles missing details
Take Brunch w Tim Canham as an example.

On jb..com it appears as `87: Tim Canham` however should actually be - as is found on Fireside - `Brunch With Brent: Tim Canham`

## Fireside: 
https://extras.show/87

image

## JB Episode page: 
https://www.jupiterbroadcasting.com/show/jupiter-extras/87/

image

## JB Show Page - Extras: 
https://www.jupiterbroadcasting.com/show/jupiter-extras/

image

## JB Home Page: 
https://www.jupiterbroadcasting.com/page/4/

image

### PR DESCRIPTION
**Pull request by ChanceM**
Implemented a more robust regex based clean of show titles. Tested against the last 10 and first 10 episode titles from each show or all 11 of Office Hours episode titles. Tested with a full scrape. 

fixes jupiterbroadcasting/jupiterbroadcasting.com#412

## Test Results
```
====================== test session starts =======================
collected 7 items                                                

test_get_plain_title.py::test_selfhosted PASSED            [ 14%]
test_get_plain_title.py::test_extras PASSED                [ 28%]
test_get_plain_title.py::test_coder PASSED                 [ 42%]
test_get_plain_title.py::test_lan PASSED                   [ 57%]
test_get_plain_title.py::test_lup PASSED                   [ 71%]
test_get_plain_title.py::test_officehours PASSED           [ 85%]
test_get_plain_title.py::test_random_cases PASSED          [100%]

======================= 7 passed in 0.00s ========================
```

## Examples of tests file
```python
import re

SHOW_TITLE_REGEX = re.compile(r"^(?:(?:Episode)?\s?[0-9]+:+\s+)?(.+?)(?:(\s+\|+.*)|\s+)?$")


def get_plain_title(title: str) -> str:
    """
    Get just the show title, without any numbering etc
    """
    return SHOW_TITLE_REGEX.match(title)[1]

def test_selfhosted():
    assert get_plain_title("78: We Should Know Better") == "We Should Know Better"
    assert get_plain_title("77: Automations Gone Wrong") == "Automations Gone Wrong"
    assert get_plain_title("76: Solid as a Rock") == "Solid as a Rock"
    assert get_plain_title("75: In-Flight Changes") == "In-Flight Changes"
    assert get_plain_title("74: A Pi For Every Problem") == "A Pi For Every Problem"
    assert get_plain_title("73: 100 Days of HomeLab") == "100 Days of HomeLab"
    assert get_plain_title("72: First Account is Free") == "First Account is Free"
    assert get_plain_title("71: Recipe for Success") == "Recipe for Success"
    assert get_plain_title("70: Plausible Deniability") == "Plausible Deniability"
    assert get_plain_title("69: Get Off My Lawn, The Robot's Got It") == "Get Off My Lawn, The Robot's Got It"

    assert get_plain_title("9: Conquering Planned Obsolescence") == "Conquering Planned Obsolescence"
    assert get_plain_title("8: WLED Changes the Game") == "WLED Changes the Game"
    assert get_plain_title("7: Why We Love Home Assistant") == "Why We Love Home Assistant"
    assert get_plain_title("6: Low Cost Home Camera System") == "Low Cost Home Camera System"
    assert get_plain_title("5: ZFS Isn’t the Only Option") == "ZFS Isn’t the Only Option"
    assert get_plain_title("4: The Joy of Plex with Elan Feingold") == "The Joy of Plex with Elan Feingold"
    assert get_plain_title("3: Home Network Under $200") == "Home Network Under $200"
    assert get_plain_title("2: Why Self-Host? With Wendell from Level1techs") == "Why Self-Host? With Wendell from Level1techs"
    assert get_plain_title("1: The First One") == "The First One"
    assert get_plain_title("Self-Hosted Coming Soon") == "Self-Hosted Coming Soon"

def test_lup():
    assert get_plain_title("474: Linux's Malware Inevitability") == "Linux's Malware Inevitability"
    assert get_plain_title("473: End of the Road") == "End of the Road"
    assert get_plain_title("472: 5 Problems With NixOS") == "5 Problems With NixOS"
    assert get_plain_title("471: The Cottonwood Disaster") == "The Cottonwood Disaster"
    assert get_plain_title("470: Let's Call It an Upgrade") == "Let's Call It an Upgrade"
    assert get_plain_title("469: Tough Linux Love") == "Tough Linux Love"
    assert get_plain_title("468: The Read Only Scenario") == "The Read Only Scenario"
    assert get_plain_title("467: All Hands on Deck") == "All Hands on Deck"
    assert get_plain_title("466: The Night of a Thousand Errors") == "The Night of a Thousand Errors"
    assert get_plain_title("465: Too Nixy for My Shirt") == "Too Nixy for My Shirt"

    assert get_plain_title("Episode 10: The Ubuntu Hangover | LINUX Unplugged 10") == "The Ubuntu Hangover"
    assert get_plain_title("Episode 9: The Ubuntu Situation | LINUX Unplugged 9") == "The Ubuntu Situation"
    assert get_plain_title("Episode 8: Cloud Guilt | LINUX Unplugged 8") == "Cloud Guilt"
    assert get_plain_title("Episode 7: Full SteamOS Ahead | LUP 7") == "Full SteamOS Ahead"
    assert get_plain_title("Episode 6: The Android Problem | LINUX Unplugged 6") == "The Android Problem"
    assert get_plain_title("Episode 5: Wrath of Linus | LINUX Unplugged 5") == "Wrath of Linus"
    assert get_plain_title("Episode 4: Are Linux Users Cheap? | LINUX Unplugged 4") == "Are Linux Users Cheap?"
    assert get_plain_title("Episode 3: Go Dock Yourself | LINUX Unplugged 3") == "Go Dock Yourself"
    assert get_plain_title("Episode 2: Edge of Failure | LINUX Unplugged 2") == "Edge of Failure"
    assert get_plain_title("Episode 1: Too Much Choice | LU1") == "Too Much Choice"
```

### CODE REVIEW

**Comment by elreydetoda** [comment]
I really like how you included test for this! it's got me thinking :thinking: ....

Whenever I try and review this over the weekend I might just have you submit those tests in the PR (or another one) as well :grin: 

There are quite a few inconsistencies with naming, so having something that we can throw corner cases out would be nice. Nice job on this! :partying_face:

**Comment by ChanceM** [comment]
I would be happy to. I sensed some trepidation to make changes to the script so I wanted to provide as much assurance as I could. Might make sense to break up the script into modules and have tests for them in the future.

**Comment by elreydetoda** [comment]
> Might make sense to break up the script into modules and have tests for them in the future.

Ya, that's something I've been wanting to do [for a while](https://github.com/JupiterBroadcasting/show-scraper/issues/20) since I'd been more acquainted with the script. Just hadn't had the time yet, because I was initially focusing on the [JB 1.0 launch](https://github.com/JupiterBroadcasting/jupiterbroadcasting.com/milestone/1), helping with [automating terraform deploys](https://github.com/JupiterBroadcasting/infra/pull/11), and then making the [E2E tests be more surefire](https://github.com/JupiterBroadcasting/jupiterbroadcasting.com/pull/353) :sweat_smile: .

**Review by gerbrent — APPROVED** [review]
This is all very impressive. Nicely done! I have zero objections, and bow to the regex master.

File: tests/test_get_plain_title.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+    assert get_plain_title("4: Finding Our Squeaky Wheels") == "Finding Our Squeaky Wheels"
+    assert get_plain_title("3: New Website Energy") == "New Website Energy"
+    assert get_plain_title("2: Podcasting Perils") == "Podcasting Perils"
+
+    assert get_plain_title("1: The Enthusiast Trap ") == "The Enthusiast Trap"
+
+def test_random_cases():
+    assert get_plain_title("1: 1: The Enthusiast Trap ") == "1: The Enthusiast Trap"
+    assert get_plain_title("1: 1:The Enthusiast Trap ") == "1:The Enthusiast Trap"
+    assert get_plain_title("Z 1:") == "Z 1:"
```
Comment (elreydetoda) [inline tests/test_get_plain_title.py:136]: I like these random cases :grin: , it give us known "fail" states (even though it's what you're expecting :sweat_smile:)

File: Pipfile [hunk 2]
```diff
@@ -17,6 +17,7 @@ pydantic-yaml = "*"
 [dev-packages]
 black = "*"
 pylint = "*"
+pytest = "*"
```
Comment (elreydetoda) [inline Pipfile:20]: looks like the lock file wasn't regenerated. Would you mind running `pipenv lock`? (just so the Pipfile.lock gets updated with the necessary hashed versions, and is repeatable across installs)

**Comment by elreydetoda** [comment]
So, I ran the scraper with the main repo and then I ran the scraper based on this branch. Afterwards I ran a diff on it (`diff -r tmp/data_main data` (ignore the sponsor change)), and it looks like these are all the affected files:

```diff
diff --color -r tmp/data_main/content/show/coder-radio/0126.md data/content/show/coder-radio/0126.md
10c10
<   "title": "Back To The Future",
---
>   "title": "HTML5: Back To The Future",
diff --color -r tmp/data_main/content/show/coder-radio/0186.md data/content/show/coder-radio/0186.md
10c10
<   "title": "Native vs Hybrid",
---
>   "title": "Decision 2016: Native vs Hybrid",
diff --color -r tmp/data_main/content/show/coder-radio/0188.md data/content/show/coder-radio/0188.md
10c10
<   "title": "Bug or Feature?",
---
>   "title": "Linux: Bug or Feature?",
diff --color -r tmp/data_main/content/show/coder-radio/0207.md data/content/show/coder-radio/0207.md
10c10
<   "title": "Too Big to Fail",
---
>   "title": "AGILE: Too Big to Fail",
diff --color -r tmp/data_main/content/show/jupiter-extras/0006.md data/content/show/jupiter-extras/0006.md
10c10
<   "title": "Chz Bacon",
---
>   "title": "Brunch with Brent: Chz Bacon",
diff --color -r tmp/data_main/content/show/jupiter-extras/0007.md data/content/show/jupiter-extras/0007.md
10c10
<   "title": "Alex Kretzschmar",
---
>   "title": "Brunch with Brent: Alex Kretzschmar",
diff --color -r tmp/data_main/content/show/jupiter-extras/0009.md data/content/show/jupiter-extras/0009.md
10c10
<   "title": "Bunk Beds",
---
>   "title": "User Error Outtake: Bunk Beds",
diff --color -r tmp/data_main/content/show/jupiter-extras/0010.md data/content/show/jupiter-extras/0010.md
10c10
<   "title": "Drew DeVore",
---
>   "title": "Brunch with Brent: Drew DeVore",
diff --color -r tmp/data_main/content/show/jupiter-extras/0012.md data/content/show/jupiter-extras/0012.md
10c10
<   "title": "Wes Payne",
---
>   "title": "Brunch with Brent: Wes Payne",
diff --color -r tmp/data_main/content/show/jupiter-extras/0018.md data/content/show/jupiter-extras/0018.md
10c10
<   "title": "Christophe Limpalair",
---
>   "title": "Brunch with Brent: Christophe Limpalair",
diff --color -r tmp/data_main/content/show/jupiter-extras/0019.md data/content/show/jupiter-extras/0019.md
10c10
<   "title": "Reverse Proxy Basics",
---
>   "title": "Self-Hosted: Reverse Proxy Basics",
diff --color -r tmp/data_main/content/show/jupiter-extras/0021.md data/content/show/jupiter-extras/0021.md
10c10
<   "title": "Angela Fisher",
---
>   "title": "Brunch with Brent: Angela Fisher",
diff --color -r tmp/data_main/content/show/jupiter-extras/0022.md data/content/show/jupiter-extras/0022.md
10c10
<   "title": "Allan Jude",
---
>   "title": "Brunch with Brent: Allan Jude",
diff --color -r tmp/data_main/content/show/jupiter-extras/0029.md data/content/show/jupiter-extras/0029.md
10c10
<   "title": "Martin Wimpress",
---
>   "title": "Brunch with Brent: Martin Wimpress",
diff --color -r tmp/data_main/content/show/jupiter-extras/0031.md data/content/show/jupiter-extras/0031.md
10c10
<   "title": "Jill Bryant Ryniker",
---
>   "title": "Brunch with Brent: Jill Bryant Ryniker",
diff --color -r tmp/data_main/content/show/jupiter-extras/0033.md data/content/show/jupiter-extras/0033.md
10c10
<   "title": "Emma Marshall",
---
>   "title": "Brunch with Brent: Emma Marshall",
diff --color -r tmp/data_main/content/show/jupiter-extras/0035.md data/content/show/jupiter-extras/0035.md
10c10
<   "title": "Jacob Roecker",
---
>   "title": "Brunch with Brent: Jacob Roecker",
diff --color -r tmp/data_main/content/show/jupiter-extras/0036.md data/content/show/jupiter-extras/0036.md
10c10
<   "title": "Rocco",
---
>   "title": "Brunch with Brent: Rocco",
diff --color -r tmp/data_main/content/show/jupiter-extras/0038.md data/content/show/jupiter-extras/0038.md
10c10
<   "title": "Alan Pope",
---
>   "title": "Brunch with Brent: Alan Pope",
diff --color -r tmp/data_main/content/show/jupiter-extras/0040.md data/content/show/jupiter-extras/0040.md
10c10
<   "title": "Jason Spisak Part 1",
---
>   "title": "Brunch with Brent: Jason Spisak Part 1",
diff --color -r tmp/data_main/content/show/jupiter-extras/0041.md data/content/show/jupiter-extras/0041.md
10c10
<   "title": "Jason Spisak Part 2",
---
>   "title": "Brunch with Brent: Jason Spisak Part 2",
diff --color -r tmp/data_main/content/show/jupiter-extras/0042.md data/content/show/jupiter-extras/0042.md
10c10
<   "title": "Catherine Kretzschmar",
---
>   "title": "Brunch with Brent: Catherine Kretzschmar",
diff --color -r tmp/data_main/content/show/jupiter-extras/0043.md data/content/show/jupiter-extras/0043.md
10c10
<   "title": "Jackie DeVore",
---
>   "title": "Brunch with Brent: Jackie DeVore",
diff --color -r tmp/data_main/content/show/jupiter-extras/0044.md data/content/show/jupiter-extras/0044.md
10c10
<   "title": "Joe Ressington",
---
>   "title": "Brunch with Brent: Joe Ressington",
diff --color -r tmp/data_main/content/show/jupiter-extras/0045.md data/content/show/jupiter-extras/0045.md
10c10
<   "title": "Fixing Brent's WiFi",
---
>   "title": "Self-Hosted: Fixing Brent's WiFi",
diff --color -r tmp/data_main/content/show/jupiter-extras/0048.md data/content/show/jupiter-extras/0048.md
10c10
<   "title": "Jim Salter",
---
>   "title": "Brunch with Brent: Jim Salter",
diff --color -r tmp/data_main/content/show/jupiter-extras/0050.md data/content/show/jupiter-extras/0050.md
10c10
<   "title": "Peter Adams Part 1",
---
>   "title": "Brunch with Brent: Peter Adams Part 1",
diff --color -r tmp/data_main/content/show/jupiter-extras/0051.md data/content/show/jupiter-extras/0051.md
10c10
<   "title": "Peter Adams Part 2",
---
>   "title": "Brunch with Brent: Peter Adams Part 2",
diff --color -r tmp/data_main/content/show/jupiter-extras/0056.md data/content/show/jupiter-extras/0056.md
10c10
<   "title": "Joe Ressington",
---
>   "title": "Podcasting Basics: Joe Ressington",
diff --color -r tmp/data_main/content/show/jupiter-extras/0057.md data/content/show/jupiter-extras/0057.md
10c10
<   "title": "Heather Ellsworth",
---
>   "title": "Brunch with Brent: Heather Ellsworth",
diff --color -r tmp/data_main/content/show/jupiter-extras/0059.md data/content/show/jupiter-extras/0059.md
10c10
<   "title": "Brandon Bruce",
---
>   "title": "Brunch with Brent: Brandon Bruce",
diff --color -r tmp/data_main/content/show/jupiter-extras/0061.md data/content/show/jupiter-extras/0061.md
10c10
<   "title": "Nuritzi Sanchez",
---
>   "title": "Brunch with Brent: Nuritzi Sanchez",
diff --color -r tmp/data_main/content/show/jupiter-extras/0063.md data/content/show/jupiter-extras/0063.md
10c10
<   "title": "Elizabeth K. Joseph",
---
>   "title": "Brunch with Brent: Elizabeth K. Joseph",
diff --color -r tmp/data_main/content/show/jupiter-extras/0064.md data/content/show/jupiter-extras/0064.md
10c10
<   "title": "LINUX Unplugged",
---
>   "title": "Behind the Scenes: LINUX Unplugged",
diff --color -r tmp/data_main/content/show/jupiter-extras/0065.md data/content/show/jupiter-extras/0065.md
10c10
<   "title": "Stuart Langridge",
---
>   "title": "Brunch with Brent: Stuart Langridge",
diff --color -r tmp/data_main/content/show/jupiter-extras/0066.md data/content/show/jupiter-extras/0066.md
10c10
<   "title": "Aleix Pol",
---
>   "title": "Brunch with Brent: Aleix Pol",
diff --color -r tmp/data_main/content/show/jupiter-extras/0067.md data/content/show/jupiter-extras/0067.md
10c10
<   "title": "What Will Change Post-virus?",
---
>   "title": "User Error: What Will Change Post-virus?",
diff --color -r tmp/data_main/content/show/jupiter-extras/0068.md data/content/show/jupiter-extras/0068.md
10c10
<   "title": "Daniel For\u00e9",
---
>   "title": "Brunch with Brent: Daniel For\u00e9",
diff --color -r tmp/data_main/content/show/jupiter-extras/0069.md data/content/show/jupiter-extras/0069.md
10c10
<   "title": "Neal Gompa",
---
>   "title": "Pagure a GitLab Alternative: Neal Gompa",
diff --color -r tmp/data_main/content/show/jupiter-extras/0071.md data/content/show/jupiter-extras/0071.md
10c10
<   "title": "Sri Ramkrishna",
---
>   "title": "Brunch with Brent: Sri Ramkrishna",
diff --color -r tmp/data_main/content/show/jupiter-extras/0073.md data/content/show/jupiter-extras/0073.md
10c10
<   "title": "Kyle Rankin",
---
>   "title": "Brunch with Brent: Kyle Rankin",
diff --color -r tmp/data_main/content/show/jupiter-extras/0074.md data/content/show/jupiter-extras/0074.md
10c10
<   "title": "Philip M\u00fcller",
---
>   "title": "Brunch with Brent: Philip M\u00fcller",
diff --color -r tmp/data_main/content/show/jupiter-extras/0075.md data/content/show/jupiter-extras/0075.md
10c10
<   "title": "Carl Richell",
---
>   "title": "Brunch with Brent: Carl Richell",
diff --color -r tmp/data_main/content/show/jupiter-extras/0076.md data/content/show/jupiter-extras/0076.md
10c10
<   "title": "Our Reaction",
---
>   "title": "Linus Tech Tips Linux Challenge: Our Reaction",
diff --color -r tmp/data_main/content/show/jupiter-extras/0086.md data/content/show/jupiter-extras/0086.md
10c10
<   "title": "Quentin Stafford-Fraser",
---
>   "title": "Brunch with Brent: Quentin Stafford-Fraser",
diff --color -r tmp/data_main/content/show/jupiter-extras/0087.md data/content/show/jupiter-extras/0087.md
10c10
<   "title": "Tim Canham",
---
>   "title": "Brunch With Brent: Tim Canham",
```

---

I think those are all expected and it seems like there weren't any regressions from current titles, only additions :grin: 

I'll have to figure out how we'll merge all these changes with the website, since the scraper is in latest only mode (which means get the most 5 recent shows). Leaving this here for now though (continuing to do my review, will post soon)

File: Makefile [hunk 3]
```diff
@@ -4,3 +4,6 @@ scrape: clean
 clean:
 	-rm -r ./data
 	-mkdir ./data
+
```
Comment (elreydetoda) [inline Makefile:7]: since we're doing things in docker normally (i.e. the `scrape` target), we should probably make a note here for making this docker compatible later on:

```suggestion

# FIXME: make docker compatible in the future
```

File: tests/test_get_plain_title.py [hunk 4]
```diff
@@ -0,0 +1,145 @@
+import re
+
+SHOW_TITLE_REGEX = re.compile(r"^(?:(?:Episode)?\s?[0-9]+:+\s+)?(.+?)(?:(\s+\|+.*)|\s+)?$")
+
+
+def get_plain_title(title: str) -> str:
+    """
+    Get just the show title, without any numbering etc
+    """
+    return SHOW_TITLE_REGEX.match(title)[1]
```
Comment (elreydetoda) [inline tests/test_get_plain_title.py]: So, my only concern with this is that we're repeating ourselves with the regex. If we make a tweak in the scraper.py file then we'd also have to implement the same change here. This is what I'd recommend doing so we can ensure consistency in our tests and "application code".

```suggestion
from scraper import get_plain_title
```

File: Makefile [hunk 5]
```diff
@@ -4,3 +4,6 @@ scrape: clean
...
+test:
+	pytest --no-header -v
```
Comment (elreydetoda) [inline Makefile]: So, I personally like the header, because it give context on different information that's loaded for this pytest run. Is there a particular reason why you ran with `--no-header`?

Also, I personally normally use `-ra` as well as the `-v`. It gives a nice little summary of things when they are anything besides passing. I learned about it via a udemy class a while back and had forgotten why I used it, but it's pretty cool IMO.

image

Lastly, since we don't currently have a project structure :sweat_smile: , the best way to facilitate [this](https://github.com/JupiterBroadcasting/show-scraper/pull/35#discussion_r970177935) would be to run pytest like this: `python -m pytest...`. 
tl;dr: we need to structure our project better to facilitate [imports natively](https://docs.pytest.org/en/7.1.x/explanation/pythonpath.html)

So, based on my feedback above I'd run it like this:

```suggestion
	python -m pytest -ra -v
```

:arrow_up: is almost like what I use for the test folder on the main jb.com website repo with pytest.ini: https://github.com/JupiterBroadcasting/jupiterbroadcasting.com/blob/7f1a2aa7402f7cfdb250dce4ea5e3208c07420dc/pytest.ini#L5

File: scraper.py [hunk 6]
```diff
@@ -69,6 +70,8 @@
 
 CHAPTERS_URL_TPL = "https://feeds.fireside.fm/{show}/json/episodes/{ep_id}/chapters"
 
+# Regex to strip Episode Numbers and information after the |
```
Comment (elreydetoda) [inline scraper.py:73]: I personally like to link to regex101.com to help people that want to know more about how the regex works.

```suggestion
# Regex to strip Episode Numbers and information after the |
# https://regex101.com/r/gkUzld/
```

If people go to PCRE2 it'll actually show you how it matches with the debugging feature at the bottom left :grin:

**Review by elreydetoda — CHANGES_REQUESTED** [review]
Overall everything looks pretty good, just a few tweaks/questions and everything will be ready for merge (IMO) :grin: 

Thank you for submitting this and sticking with it @ChanceM! :partying_face: :tada:

File: Makefile (hunk 5 above)
Comment (ChanceM) [reply Makefile]: :shrug: I'm a minimalist at heart, just enough info to show what was tested and if it passed.

**Review by elreydetoda — APPROVED** [review]
LGTM (I can't merge in this repo, so we have to wait for @gerbrent to do that)

**Comment by gerbrent** [comment]
all above looks good!

**Comment by gerbrent** [comment]
@ChanceM: just needing to implement the changes requested by @elreydetoda then looks good to merge.

**Comment by ChanceM** [comment]
@gerbrent I believe all the changes where made in commit #[5003a2a](https://github.com/JupiterBroadcasting/show-scraper/pull/35/commits/5003a2af0298218d8929e45563d7879d4baeb6e9) and #[010457a](https://github.com/JupiterBroadcasting/show-scraper/pull/35/commits/010457aa3381475835aa09db0e7bf2bb0c816e60)

**Comment by elreydetoda** [comment]
Yep @gerbrent , @ChanceM is right. He's already implemented the suggestions and it is ready to merge 🥳

**Comment by elreydetoda** [comment]
Looks like @gerbrent gave me write permissions to the repo now :grin: 
So, I'll go ahead and merge this :partying_face:

**Comment by elreydetoda** [comment]
So...after merging this the scraper failed after I kicked off a manual job for it (to test and see if there were any issues) ... : https://github.com/JupiterBroadcasting/jupiterbroadcasting.com/actions/runs/3138353438/attempts/2

It seems like it doesn't like the updated hashes for your Pipfile.lock... I started testing it locally and I recieved this after doing a `pipenv --rm && pipenv install -d`. After running a `pipenv lock` on my machine it seems like it installed some different hashes for some reason :thinking: 

image


Looking at your Pipfile.lock it actually looks like it installed and locked on an [older version pydantic-yaml](https://github.com/JupiterBroadcasting/show-scraper/pull/35/files#diff-a86c67a0a29ed0e95909b9b7c420140f302d17399ee6dcce4e1a51a14d27fd51R192)... :thinking: 

I don't know why or how this happened, but I'm going to push my lock file and see if it helps.

**Comment by elreydetoda** [comment]
alright, seems like that fixed it: https://github.com/JupiterBroadcasting/jupiterbroadcasting.com/actions/runs/3138353438

yay for it working now that :sweat_smile:

**Comment by elreydetoda** [comment]
Perfect example of this implementation working! Nicely done @ChanceM :partying_face: 

https://github.com/JupiterBroadcasting/jupiterbroadcasting.com/commit/96c2bf3ac52b1c46b436d909f15e753cd0876ed0
